### ISSUE TITLE
TriblerRequestManager is unsafe when not bound to a variable

### ISSUE THREAD

**Issue by ichorid** [Tribler/tribler#4606]
This kind of code leads to misrouted responses:

```python
mgr = TriblerRequestManager()
mgr.perform_request("/some_url", first_callback)

mgr = TriblerRequestManager()
mgr.perform_request("/some_other_url", second_callback)
```
After calls like these, one can't tell which callback will be really called for which request.
Is it by design? Am I using it wrong?

**Comment by qstokkink** [comment]
Should we do this structural change for 7.3 already? We might want to wait until later date.

**Comment by ichorid** [comment]
Closed by #5021. RequestManager is a singleton now.
